src/eth/erc4337/types.ts

```diff
@@ -1,4 +1,4 @@
-import { string, type Infer } from 'superstruct';
+import { optional, string, type Infer } from 'superstruct';
```

Comment:
You should use `exactOptional` from `../../superstruct` instead of `optional` to correctly handle the `exactOptionalPropertyTypes` compiler flag.